### DISCUSSION TITLE
The Github Mobile App 2FA isn't working

### DISCUSSION THREAD

**Discussion by HCLarsen** [community/community#69171] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I'm trying to set up 2FA using the Github mobile app, which is an option based on your own blog site.

<img width="718" alt="Screenshot 2023-10-03 at 11 41 12 PM" src="https://github.com/community/community/assets/8650607/bac86195-1144-46e7-a5b9-46c506101e80">

When I try this, I enter the 2 digit code, and then the next screen on my computer asks me to scan the QR code. However, there's no scan option in the Github Mobile App, so I have to hit cancel.

Can anyone make sense of this?

**Comment by radar-blip** [comment]


**Reply by HCLarsen** [reply]
It should be obvious that I don't want to use a different method, since I'm looking for help with this one.

Also, SMS based 2FA is easy to hack with methods like Sim Card Hacking, making it less secure than no 2FA, so you really shouldn't suggest that people do that. No one should.

**Comment by DNin01** [comment]
GitHub Mobile should already be set up - if I recall correctly, it is essentially set up automatically by having your mobile devices logged via GitHub Mobile. You can test it out by logging out and logging back in or by doing something that requires authentication (something that pops up the sudo prompt like creating/adjusting a ruleset or viewing recovery codes).

From what you explained, you might be trying to reconfigure your authenticator app (Settings → Password and authentication → Authenticator app → Edit) which explains why you're getting the QR Code, but I don't think I've ever had to set up authentication via GitHub Mobile (it has just worked out of the box), so I don't know.

**Reply by HCLarsen** [reply]
`From what you explained, you might be trying to reconfigure your authenticator app `

I very clearly stated that what I was trying to do was set up 2FA.

I've since given up trying since Github doesn't seem to care.

**Reply by DNin01** [reply]
> `From what you explained, you might be trying to reconfigure your authenticator app `
> 
> 
> 
> I very clearly stated that what I was trying to do was set up 2FA.

Oh, so you don't have any form of 2FA enabled right now, is that what you meant?

If so, GitHub Mobile is not one of the available options during initial 2FA setup - you must configure either a TOTP app such as Google Authenticator, or SMS (which you did not want), then you will be able to use GitHub Mobile to authenticate in addition to the TOTP app.

If not, take a look at the "GitHub Mobile" option in the 2FA settings page. If it says "Configured", you're all set up.

**Reply by HCLarsen** [reply]
> If so, GitHub Mobile is not one of the available options during initial 2FA setup - you must configure either a TOTP app such as Google Authenticator, or SMS (which you did not want), then you will be able to use GitHub Mobile to authenticate in addition to the TOTP app.

The screenshot I posted above, from Github's own documentation, clearly states that it is an option.

**Reply by DNin01** [reply]
> > If so, GitHub Mobile is not one of the available options during initial 2FA setup - you must configure either a TOTP app such as Google Authenticator, or SMS (which you did not want), then you will be able to use GitHub Mobile to authenticate in addition to the TOTP app.
> 
> The screenshot I posted above, from Github's own documentation, clearly states that it is an option.

A preferred 2FA method just means which method is prompted for use by default. You still have to start out by setting up either a TOTP app or SMS, and have them configured for as long as 2FA is enabled. This is officially stated in the [documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication):
> You can configure two-factor authentication (2FA) using a TOTP app on mobile or desktop or via text message. After you have configured 2FA using a TOTP app or via text message, you can then also add security keys as alternate 2FA methods.

> Once you have configured a TOTP application, or SMS, you can also use GitHub Mobile to authenticate. If, in the future, you no longer have access to GitHub Mobile, you will still be able to use security keys or TOTP applications to sign in.

Oh, and that's also why they don't let you set up GitHub Mobile for 2FA initially. Since GitHub Mobile 2FA relies on having the sign-in request sent to your account and picked up by your signed-in mobile device(s), if you signed out of GitHub Mobile, you wouldn't have that as a 2FA option and it wouldn't be possible to sign back into it again in order to authenticate with it, causing a lockout. Requiring TOTP or SMS to be configured prevents that from happening.

**Reply by HCLarsen** [reply]
> A preferred 2FA method just means which method is prompted for use by default. You still have to start out by setting up either a TOTP app or SMS, and have them configured for as long as 2FA is enabled.

It makes no sense at all to require one form of 2FA to enable another. It just further shows how Github is failing its users.

In any case, as I've already pointed out, this is a closed issue, since I've moved on, over a week ago. There's no point at all in discussing this further.

**Comment by zlice** [comment]
Other 2fa doesn't seem to work either. At least for me. Keep getting locked out, IP temp-bans, even recovery codes didn't work a few this morning.

So basically...I'm going to be shadow-banned from contributing to projects, most of which are hosted on github.
